### ISSUE TITLE
Refactor user answer logs and create new collection for faster progress reporting.

### ISSUE THREAD

**Issue by andrewtackett** [memphis-iis/mofacts-ies#64]
Currently studentProgressReporting.js replays the entire userTimesLog collection to calculate the user progress.  Add in additional collection and append to it when users answer to make calculation of user progress faster.

**Comment by wscarter** [comment]
Also to refactor:  instructorReporting.js should be refactored to use the same (backend) calculation method that studentReporting uses to get individual reports.

**Comment by imrryr** [comment]
yes, isn't this effectively the same item as the database upgrade? or can we do anything before that?

**Comment by imrryr** [comment]
Renamed since I can't see working on the current database if we are replacing it.

**Comment by imrryr** [comment]
Please review whether this choice is optimal https://www.postgresql.org/ to begin

I did a bit of research...

**Comment by andrewtackett** [comment]
@imrryr As Wes said over email, we have chosen postgres for our standard tech stack so it would fit in well.  Most major SQL databases would actually, but whether it is the best ultimate choice will require research into how we can substitute in the new database (I've done some preliminary research and there aren't a lot of meteor frameworks for doing this but there are some).

**Comment by andrewtackett** [comment]
Staged approach/stretch goals may be best.  Primary focus is migrating userTimesLog's and allowing faster engine state restoration instead of manually replaying every utl row.  Secondary focus would be everything else but auth, which is tied into meteor.  Third focus would be reimplementing auth via the meteor accounts api, possibly by creating a new login handler a la the existing SAML custom package except with direct db calls for local user/pass.

**Comment by andrewtackett** [comment]
@imrryr My plan for efficiency improvements is essentially the same as above, first make sure we can do a transfer of userTimesLog's to Postgres with no structure change, then make incremental improvements from there.

**Comment by andrewtackett** [comment]
Main work on this is done.  Bugs can be worked on in groups/in new issues.
